abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1996 @@
+
+Adds key (`key_data`, `key_size`) with value (`value_data`,
+`value_size`) to the header map `map_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`, `value_data`
+  and/or `value_size` point to invalid memory address.
+
+
+#### `proxy_replace_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_header_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const uint8_t *) value_data`
+  - `i32 (size_t) value_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Adds or replaces key's (`key_data`, `key_value`) value with the provided
+value (`value_data`, `value_size`) in the header map `map_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`, `value_data`
+  and/or `value_size` point to invalid memory address.
+
+
+#### `proxy_remove_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_header_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Removes the key (`key_data`, `key_value`) from the header map `map_id`.
+
+Returned `status` value is:
+- `OK` on success (including the case when requested key didn't exist).
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data` and/or `key_size` point to
+  invalid memory address.
+
+
+## Common stream operations
+
+### Functions exposed by the host
+
+#### `proxy_continue_stream`
+
+* params:
+  - `i32 (`[`proxy_stream_type_t`]`) stream_type`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Resumes processing of paused `stream_type`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `stream_type`.
+- `UNIMPLEMENTED` when continuation of requested `stream_type`
+  is not supported.
+
+
+#### `proxy_close_stream`
+
+* params:
+  - `i32 (`[`proxy_stream_type_t`]`) stream_type`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Closes or resets `stream_type`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `stream_type`.
+
+
+#### `proxy_get_status`
+
+* params:
+  - `i32 (uint32_t *) return_status_code`
+  - `i32 (const char **) return_status_message_data`
+  - `i32 (size_t *) return_status_message_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Retrieves status code (`return_status_code`) and status message
+(`return_status_message_data`, `return_status_message_size`) of
+the HTTP call when called from [`proxy_on_http_call_response`]
+or gRPC stream or call when called from [`proxy_on_grpc_close`].
+
+Returned `status` value is:
+- `OK` on success.
+- `INVALID_MEMORY_ACCESS` when `return_status_code`,
+  `return_status_message_data` and/or `return_status_message_size`
+  point to invalid memory address.
+
+
+## TCP/UDP/QUIC streams
```

Comment:
I think we should limit this to only `TCP streams` since UDP/QUIC are not supposed in v0.2.1.